src/runtime/polyfill/cloudflare.ts

```diff
@@ -0,0 +1,4 @@
+import { nextTick } from "node:process";
+import { process as _process } from "../node/process/_process";
+
+Object.assign(_process, { nextTick });
```

Comment:
And one more! We could also _prefer_ (if available) `queueMicrotask` in [main process polyfill](https://github.com/unjs/unenv/blob/main/src/runtime/node/process/_process.ts#L137) with an early return. This makes the fix generic for all compatible runtimes.

Reply:
> And one more! We could also prefer (if available) queueMicrotask in main process polyfill with an early return. This makes the fix generic for all compatible runtimes.

That polyfill relies on setTimeout to schedule the work which is not the same as `queueMicrotask` — microtasks take precedence over tasks, so the current polyfill will run the callbacks in incorrect order.

Is there any reason for the nextTick polyfill not to use queueMicrotask? It seems to be widely supported API: https://developer.mozilla.org/en-US/docs/Web/API/queueMicrotask

> would it make sense that we avoid node: imports or do you see that in the future workerd might do it differently?

We do expect the workerd runtime to get a few more (process) APIs polyfilled over time, but more importantly this PR is an attempt to create a hybrid polyfill at a symbol level, similarly how https://github.com/unjs/unenv/issues/181 proposes it for the module level. We want to prove that this is possible, as that would then inform our overall node compat strategy.

Reply:
No reason main polyfill not using `queueMicrotask` it was simply remaining from the legacy polyfills I used. 

We can use a condition in there to use `queueMicrotask` when available and eventually drop fallback if most of the runtimes support it.

> We want to prove that this is possible, as that would then inform our overall node compat strategy.

If it is helpful for your testing strategy, looks good to me this way 👍🏼 

From https://github.com/unjs/unenv/issues/181#issuecomment-2110447976 I was under the impression that you would want to go with `process.getBuiltinModule` API and that this PR particularly can be natively improved in main unenv's `nextTick` polyfill.

Reply:
@pi0 

I ended up importing `node:process` this way because I couldn't find another way to grab the native workerd global `process` in a reliable way. Is there a better way of doing it?

These all didn't seem to work:

```ts
console.log(process); //undefined
console.log(globalThis.process); // undefined
console.log(global.process); // undefined
```

Reply:
@jculvey the issue here is that workerd doesn't expose the process object on global at all (unless you use the underutilized `nodeJsCompatModule`, which we want to avoid because we are thinking of removing it).

Reply:
globalThis.process and `node:process` seem to have the same shape however, with the exception of `node:process#default` which exists to support default import.

Reply:
Right, my mistake. Thanks!